system/include/emscripten/val.h

```diff
@@ -419,16 +419,17 @@ namespace emscripten {
             return MethodCaller<ReturnValue, Args...>::call(handle, name, std::forward<Args>(args)...);
         }
 
-        template<typename T>
-        T as() const {
+        template<typename T, typename ...Policies>
+        T as(Policies...) const {
             using namespace internal;
 
             typedef BindingType<T> BT;
+            typename WithPolicies<Policies...>::template ArgTypeList<T> targetType;
 
             EM_DESTRUCTORS destructors;
             EM_GENERIC_WIRE_TYPE result = _emval_as(
                 handle,
-                TypeID<T>::get(),
+                targetType.getTypes()[0],
```

Comment:
Here, I think calling `[0]` should be fine since `ArgTypeList` l.427 received at least one type argument.